### PR TITLE
Add Nullability annotations to public API

### PR DESCRIPTION
**Pull request by rhyshort**
## What
Add Nullability annotation to public API to improve interoperability
with swift.

## How
Add nullability specifiers to classes which users are expected to interact with. 
In addition to this, collections have been made generic if required and possible to do  #so.

## Reviewers
reviewer @alfinkel  
## Issues

bugzId: 48965

### CODE REVIEW

File: CDTDatastore/CDTDatastore.h [hunk 1]
```diff
@@ -182,7 +182,7 @@ extern NSString *const CDTDatastoreChangeNotification;
  * Older revisions will not contain the document data as it will have
  * been compacted away.
  */
-- (NSArray *)getRevisionHistory:(CDTDocumentRevision *)revision;
+- (nonnull NSArray<CDTDocumentRevision*>  *)getRevisionHistory:(nonnull CDTDocumentRevision *)revision;
```
Comment (alfinkel) [inline CDTDatastore/CDTDatastore.h]: There appears to be an extra space after `<CDTDocumentRevision*>`.

**Comment by alfinkel** [comment]
+1 after whitespace is addressed.

File: CDTDatastore/CDTDatastore.h (hunk 1 above)
Comment (rhyshort) [inline CDTDatastore/CDTDatastore.h]: fixed in c90a8e4

**Comment by rhyshort** [comment]
The failures are encryption related and since travis passes the checks, I'm going to go ahead and merge

**Pull request merged**